src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/Groups.scala

```diff
@@ -93,10 +94,22 @@ class Groups[F[_]](ref: ActorRef)(implicit cfg: GroupsConfig, hc: HttpClient[F,
       cursor.get[String]("groups").map(_.split(",").map(_.trim).filterNot(_.isEmpty).toSet)
 
     val req = Get(realm.userInfoEndpoint.asUri).addCredentials(OAuth2BearerToken(token.value))
-    hc(req).map { json =>
-      val stringGroups = fromSet(json.hcursor) orElse fromCsv(json.hcursor) getOrElse Set.empty[String]
-      stringGroups.map(str => Group(str, realm.id.value))
-    }
+    hc(req)
+      .map { json =>
+        val stringGroups = fromSet(json.hcursor) orElse fromCsv(json.hcursor) getOrElse Set.empty[String]
+        stringGroups.map(str => Group(str, realm.id.value))
+      }
+      .recoverWith {
+        case UnexpectedUnsuccessfulHttpResponse(resp, body) =>
+          if (resp.status == StatusCodes.Unauthorized || resp.status == StatusCodes.Forbidden) {
+            logger.warn(s"A provided client token was rejected by the OIDC provider, reason: '$body'")
+            F.raiseError(IamError.InvalidAccessToken(TokenRejection.InvalidAccessToken))
+          } else {
+            logger.warn(
+              s"A call to get the groups from the OIDC provider failed unexpectedly, status '${resp.status}', reason: '$body'")
+            F.raiseError(IamError.InternalError("Unable to extract group information from the OIDC provider."))
```

Comment:
This will be logged twice, once here and another in the [ExceptionHandling](https://github.com/BlueBrain/nexus-iam/blob/master/src/main/scala/ch/epfl/bluebrain/nexus/iam/routes/Routes.scala#L54). You could avoid the double logging by just doing
```scala
else F.raiseError(IamError.InternalError(s"A call to get the groups from the OIDC provider failed unexpectedly, status '${resp.status}', reason: '$body'")))
```

Reply:
Can't really do that because the message would be presented to the client, but the information is still valid in the log.